src/python/pants/backend/python/framework/stevedore/target_types.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.engine.target import StringSequenceField, Targets
+from pants.util.strutil import softwrap
+
+
+class StevedoreNamespace(str):
+    """Syntactic sugar to tag a namespace in entry_points as a stevedore namespace.
+
+    For example:
+        python_distribution(
+            ...
+            entry_points={
+                stevedore_namespace("a.b.c"): {
+                    "plugin_name": "some.entry:point",
+                },
+            },
+        )
+    """
+
+    alias = "stevedore_namespace"
+
+
+# This is a lot like a SpecialCasedDependencies field, but it doesn't list targets directly.
+class StevedoreNamespacesField(StringSequenceField):
+    alias = "stevedore_namespaces"
```

Comment:
Hmm. Do I need to include an `experimental_` prefix?
```suggestion
    alias = "experimental_stevedore_namespaces"
```

Same for the `stevedore_namespace` object?

Reply:
I think it may be enough to have the experimental tag in the backend name. In cases where the targets are prefixed with `experimental_` it's either being added to a non-experimental backend or that particular target is very explorative. That's my take on it at least :)

Reply:
great. I would hate to make these overly verbose unless I had to.